policyengine_us/variables/gov/states/al/tax/income/al_medical_deduction.py

```diff
@@ -0,0 +1,20 @@
+from policyengine_us.model_api import *
+
+
+class al_medical_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Alabama Medical Deduction"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://www.revenue.alabama.gov/ultraviewer/viewer/basic_viewer/index.html?form=2023/01/22f40schabdc_blk.pdf"
+        "https://www.revenue.alabama.gov/ultraviewer/viewer/basic_viewer/index.html?form=2023/02/22f40.pdf"
+    )
+    defined_for = StateCode.AL
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.al.tax.income.deductions.itemized
+        medical_deduction = tax_unit("al_medical_dedcution", period)
```

Comment:
this is a circular reference? 

    def formula(tax_unit, period, parameters):
        expense = add(tax_unit, period, ["medical_expense"])
        medical = parameters(period).gov. - add alabama parameter path to the 4% 
        medical_floor = medical.floor * tax_unit("positive_agi", period)
        return max_(0, expense - medical_floor)